include/algorithms/multi_class_classifier/multi_class_classifier_model_builder.h

```diff
@@ -82,11 +82,12 @@ class DAAL_EXPORT ModelBuilder
      */
     void setTwoClassClassifierModel(size_t negativeClassIdx, size_t positiveClassIdx, const classifier::ModelPtr& model)
     {
-        if(negativeClassIdx >= positiveClassIdx)
+        if(negativeClassIdx >= positiveClassIdx || !_s)
```

Comment:
if status is not ok
seems we have to update status with 'ErrorMemoryAllocationFailed'
we have to differentiate checks